### DISCUSSION TITLE
Get Message from another Channel by user defined metadata

### DISCUSSION THREAD

**Discussion by MichaelLeeHobbs** [nextgenhealthcare/connect#4485] [Feature/Enhancement Ideas]
Use case: I have to store messages in a DB for them to only be used once or twice and because I have no other way to access them from another channel.

Current
1. ORM Order message comes in and is stored in DB and forwarded.
2. ORU Report message comes from PACS.
3. Transformer pulls stored ORM message out of DB.
4. Transformer combines data from ORU and ORM to send final ORU message to customer.

Proposed
1. ORM Order message comes in, metadata map k/v generated,  and forwarded.
2. ORU Report message comes from PACS.
3. Transformer pulls stored ORM message out of ORM channel with metadata.
4. Transformer combines data from ORU and ORM to send final ORU message to customer.

Would also need to look at adding indexes to metadata columns.

**Comment by leovander** [comment]
@MichaelLeeHobbs If you could report back an attachment id/channel id to PACS, you could theoretically use the attachment handler to fetch the message from another channel that was already processed. The biggest caveat being that your pruning settings need to be matched/tweaked between the two channels, since you could reference a message that has since been pruned.

**Reply by leovander** [reply]
I believe metadata columns do get indexed when you create custom ones.

**Reply by MichaelLeeHobbs** [reply]
As of 3.7.0 I checked the Mirth DB and there are no indexes on metadata columns.

**Reply by tonygermano** [reply]
The indexes are not created by default because they would slow down writes, and mirth is very write heavy.

**Comment by jonbartels** [comment]
Is it correct to think of your use case like a persistent cache that is easily accessible from MC?

**Reply by MichaelLeeHobbs** [reply]
More like a temporary cache ie messages are not needed beyond their prune date.
